test/integration_test.py

```diff
@@ -723,34 +738,73 @@ def _assert_indices_exist(self, catalog: CatalogName):
         assert isinstance(plugin, tdr.Plugin)
         # Uses the indexer service account credentials, which should have access
         # to all sources.
         tdr_client = plugin.tdr
-        all_sources = {
+        configured_sources = {
             frozendict(sourceSpec=str(source_spec),
                        sourceId=tdr_client.lookup_source_id(source_spec))
             for source_spec in map(TDRSourceSpec.parse, config.tdr_sources(catalog))
         }
-        self.assertEqual(_list_sources(tdr_client._http_client), all_sources)
+        public_sources = self._list_sources(catalog)
 
-        public_tdr_client = TDRClient.with_public_service_account_credentials()
-        unauthenticated_sources = _list_sources(http_client())
-        self.assertEqual(unauthenticated_sources,
-                         _list_sources(public_tdr_client._http_client))
-        self.assertTrue(unauthenticated_sources <= all_sources)
+        with self._http_context(self._service_account_authorized_http):
+            self.assertEqual(self._list_sources(catalog), configured_sources)
+        with self._http_context(self._public_service_account_authorized_http):
+            self.assertEqual(self._list_sources(catalog), public_sources)
+        self.assertEqual(public_sources - configured_sources, set())
+
+        managed_access_source_ids = {
+            source['sourceId']
+            for source in configured_sources - public_sources
+        }
+
+        def _source_ids_from_hits(hits: JSONs) -> Set[str]:
+            return {one(bundle['sources'])['sourceId'] for bundle in hits}
+
+        hits = self._get_entities(catalog, 'bundles')
+        indexed_source_ids = _source_ids_from_hits(hits)
+        self.assertEqual(indexed_source_ids,
+                         {source['sourceId'] for source in public_sources})
+        source_filter = {'sourceId': {'is': list(managed_access_source_ids)}}
+        url = furl(config.service_endpoint(),
+                   path='index/bundles',
+                   args={
+                       'catalog': catalog,
+                       'filters': json.dumps(source_filter)
+                   }).url
+        response = self._get_url_json(url)
+        self.assertEqual(response['hits'], [])
+        self.assertEqual(response['pagination']['total'], 0)
+
+        with self._http_context(self._service_account_authorized_http):
+            response = self._get_url_json(url)
+        indexed_source_ids = _source_ids_from_hits(response['hits'])
+        self.assertEqual(indexed_source_ids, managed_access_source_ids)
+        managed_access_files = (
+            file['url']
+            for bundle in response['hits']
+            for file in bundle['files']
+        )
+        try:
+            file_url = first(managed_access_files)
+        except ValueError:
+            pass
+        else:
+            response = self._http.request('GET', file_url)
+            self.assertEqual(response.status, 404)
```

Comment:
Would it make sense to try this again with an authorized HTTP client?

Reply:
I don't think so because `/repository/files/` doesn't support HEAD and we don't want to accidentally download a huge file (and the smallest file in `hca_dev_20210621_managedaccess_4298b4de92f34cbbbbfe5bc11b8c2422__20210622` is over 3.9 GB).

Reply:
With https://github.com/DataBiosphere/azul/issues/3245 reverted, this test is no longer applicable, but I'll do this either 1. on this PR if the replacement PR for #3245 is merged first 2. in that PR if this one is merged first